### ISSUE TITLE
Option to Include Heading in Fold

### ISSUE THREAD

**Issue by noctuid** [preservim/vim-markdown#130]
I like having title in the fold text. Instead of having the heading followed by the fold, I'd like to have the heading included (as if it was `## heading {{{` for a marker equivalent). nelstrom's vim-markdown-folding folds like this I believe, but it stopped working for me, and this plugin provides other functionality that I use.

**Comment by auvmacintosh** [comment]
I think this will be really helpful. For example I have a long memo that contains about 100 1st level headings. With this feature, the memo will be initially displayed like this:
```
+-- 21 lines: # topic 1
+-- 58 lines: # topic 2
+-- 12 lines: # topic 3
```
However at present the vim-markdown will display the memo like: 
```
# topic 1
+-- 20 lines: ...
# topic 2
+-- 57 lines: ...
# topic 3
+-- 11 lines: ...
```
The extra lines are distracted when the memo is very long.

**Comment by amfl** [comment]
Isn't this just pythonic style?

``` vim
let g:vim_markdown_folding_style_pythonic = 1
```

Gives folds like

```
191 ## LVM ------------------------------------------------------ 8---
200 ## Notes ---------------------------------------------------- 8---
209 ## Mozilla Overlay ----------------------------------------- 11---
```

I think this issue can be closed.

**Comment by Joedang** [comment]
@afml No, that's only true for H2 and smaller. In documents where you have lots of H1 headings -- especially when there's a bare paragraph under the H1 -- they pythonic folding leaves a lot of clutter. Right now, you can turn off pythonic folding and set the fold level to zero, but then you're stuck with regular folding which doesn't facilitate walking down the heading tree of a document.

IMO, it doesn't make sense to have this exception for H1s, the way it currently is. An H1 is not a title. Period. That's what the `title: ` option is for, if you're using Pandoc to render your Markdown.
